adapter/mcstoriesAdapter.py

```diff
@@ -0,0 +1,182 @@
+import re
+import urllib.parse
+from typing import Optional
+
+from bs4 import BeautifulSoup
+
+import scrape
+import util
+from adapter.adapter import Adapter, edumpContent
+from htypes import FicType, FicId
+from schema import OilTimestamp
+from store import Fic, Language
+from store import FicStatus
+
+story_index_path_re = re.compile(r"/(?P<lid>\w+)/index\.html")
+story_chapter_path_re = re.compile(r"/(?P<lid>\w+)/(?P<clid>\w+)\.html")
+relative_chapter_path_re = re.compile(r"(?P<clid>\w+)\.html")
+
+
+def strip_nonnumeric(s):
+    return ''.join(c for c in s if c.isdigit())
+
+
+class McStoriesAdapter(Adapter):
+    def __init__(self) -> None:
+        super().__init__(
+            True,
+            "https://mcstories.com/",
+            "mcstories.com",
+            FicType.mcstories,
+            "mcstories",
+        )
+
+    def tryParseUrl(self, url: str) -> Optional[FicId]:
+        path = urllib.parse.urlparse(url).path
+
+        if m := story_index_path_re.fullmatch(path):
+            lid = m.group("lid")
+        elif m := story_chapter_path_re.fullmatch(path):
+            lid = m.group("clid")
```

Comment:
The `clid` group is only the same as `lid` for stories with a single chapter. So far every multi-chapter story I've checked switches to `/{lid}/{lid}{cid}.html` format.

I haven't caught a chapter that went from one to multiple chapters, but it also seems like `/{lid}/{lid}.html` gets a 404 at that point -- which maybe isn't a problem but something to be aware of.